src/Form/SettingsForm.php

```diff
@@ -0,0 +1,271 @@
+        '#title' => $this->t('Certificate locator type'),
+        '#options' => [
+          'azure_key_vault' => $this->t('Azure key vault'),
+          'file_system' => $this->t('File system'),
+        ],
+        '#default_value' => $defaultValues[self::CERTIFICATE]['locator_type'] ?? NULL,
+      ],
+    ];
+
+    $form[self::CERTIFICATE][CertificateLocatorHelper::LOCATOR_TYPE_AZURE_KEY_VAULT] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Azure key vault'),
+      '#states' => [
+        'visible' => [':input[name="certificate[locator_type]"]' => ['value' => CertificateLocatorHelper::LOCATOR_TYPE_AZURE_KEY_VAULT]],
+      ],
+    ];
+
+    $settings = [
+      'tenant_id' => ['title' => $this->t('Tenant id')],
+      'application_id' => ['title' => $this->t('Application id')],
+      'client_secret' => ['title' => $this->t('Client secret')],
+      'name' => ['title' => $this->t('Name')],
+      'secret' => ['title' => $this->t('Secret')],
+      'version' => ['title' => $this->t('Version')],
+    ];
+
+    foreach ($settings as $key => $info) {
+      $form[self::CERTIFICATE][CertificateLocatorHelper::LOCATOR_TYPE_AZURE_KEY_VAULT][$key] = [
+        '#type' => 'textfield',
+        '#title' => $info['title'],
+        '#default_value' => $defaultValues[self::CERTIFICATE][CertificateLocatorHelper::LOCATOR_TYPE_AZURE_KEY_VAULT][$key] ?? NULL,
+        '#states' => [
+          'required' => [':input[name="certificate[locator_type]"]' => ['value' => CertificateLocatorHelper::LOCATOR_TYPE_AZURE_KEY_VAULT]],
+        ],
+      ];
+    }
+
+    $form[self::CERTIFICATE][CertificateLocatorHelper::LOCATOR_TYPE_FILE_SYSTEM] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('File system'),
+      '#states' => [
+        'visible' => [':input[name="certificate[locator_type]"]' => ['value' => CertificateLocatorHelper::LOCATOR_TYPE_FILE_SYSTEM]],
+      ],
+
+      'path' => [
+        '#type' => 'textfield',
+        '#title' => $this->t('Path'),
+        '#default_value' => $defaultValues[self::CERTIFICATE][CertificateLocatorHelper::LOCATOR_TYPE_FILE_SYSTEM]['path'] ?? NULL,
+        '#states' => [
+          'required' => [':input[name="certificate[locator_type]"]' => ['value' => CertificateLocatorHelper::LOCATOR_TYPE_FILE_SYSTEM]],
+        ],
+      ],
+    ];
+
+    $form[self::CERTIFICATE]['passphrase'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Passphrase'),
+      '#default_value' => $defaultValues[self::CERTIFICATE]['passphrase'] ?? NULL,
+    ];
+
+    $form[self::CACHE_EXPIRATION] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Cache expiration modifier'),
+      '#required' => TRUE,
+      '#default_value' => $defaultValues[self::CACHE_EXPIRATION] ?? NULL,
+      '#description' => $this->t('Should be in GNU date input format, e.g. "7am, tomorrow 7am". If multiple are provided, they should be separated by comma, and the first upcoming one is used.'),
```

Comment:
Separating with newline will be easier to read and write.

Reply:
Sure, fixed.